### PR TITLE
blaze-lib 2.4 (new formula)

### PR DESCRIPTION
**Pull request by hgossler**
Open-source, high-performance C++ math library for dense and sparse
arithmetic

### CODE REVIEW

File: blaze-lib.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class BlazeLib < Formula
+  desc "High-performance C++ math library for dense and sparse arithmetic"
+  homepage "https://bitbucket.org/blaze-lib/blaze/"
+  url "https://bitbucket.org/blaze-lib/blaze/downloads/blaze-2.4.tar.gz"
+  sha256 "34af70c8bb4da5fd0017b7c47e5efbfef9aadbabc5aae416582901a4059d1fa3"
+
+  depends_on "boost"
+
+  def install
+    inreplace "Configfile", "CXX=\"g++\"", "CXX=\"#{ENV.cxx}\""
```
Comment (dpo) [inline blaze-lib.rb]: If this library only builds with `g++`, there should be a `fails_with` block before `install` so Homebrew can use the appropriate compiler.

File: blaze-lib.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    system "./configure"
+    system "make"
+
+    include.install "blaze"
```
Comment (dpo) [inline blaze-lib.rb]: Is there no `make install`? What about documentation?

File: blaze-lib.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    lib.install "lib/libblaze.a"
+  end
+
+  test do
+    (testpath/"test.cpp").write <<-EOS.undent
+      #include <blaze/Blaze.h>
+      int main(int argc, char *argv[])
+      {
+        return 0;
+      }
```
Comment (dpo) [inline blaze-lib.rb]: This is the right way to write a test, but how about a slightly more compelling example, such as https://bitbucket.org/blaze-lib/blaze/wiki/Getting_Started#!a-first-example?

**Comment by hgossler** [comment]
Improved the test according to your suggestion. There is no make install, hence the manual copy of the files. The documentation is a separate download

**Comment by dunn** [comment]
Just a nitpick from the audit, now:
```
==> audit problems
homebrew/science/blaze-lib:
 * Description is too long. "name: desc" should be less than 80 characters.
Length is calculated as homebrew/science/blaze-lib + desc. (currently 93)
```

**Comment by hgossler** [comment]
Ok, shortened it a bit

**Comment by dpo** [comment]
Thanks! Looks good.

**Pull request closed without merging**